hpx/parallel/algorithms/for_each.hpp

```diff
@@ -45,11 +46,33 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
             typename hpx::util::decay<F>::type& f_;
             typename hpx::util::decay<Proj>::type& proj_;
 
+            template <typename T>
+            HPX_HOST_DEVICE HPX_FORCEINLINE
+            typename std::enable_if<
+               !hpx::traits::is_value_proxy<T>::value
+            >::type
+            call(T && t)
+            {
+                hpx::util::invoke(f_,
+                    hpx::util::invoke(proj_, std::forward<T>(t)));
+            }
+
+            template <typename T>
+            HPX_HOST_DEVICE HPX_FORCEINLINE
+            typename std::enable_if<
+                hpx::traits::is_value_proxy<T>::value
+            >::type
+            call(T && t)
+            {
+                auto tmp = hpx::util::invoke(proj_, std::forward<T>(t));
```

Comment:
@hkaiser, It was that. Problem solved by modifying a little bit the following file value_proxy.hpp. Done in b961aba635572e45949869c0a5485ee79b4dd2c6